### PR TITLE
UAR-1326 Add a new 'is_remove' field to the OE session data

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-1326

### Change description

* Field data also sent to API
* Transaction and OE submission created on 'presenter' screen if remove journey

Note: No unit tests added yet - still in draft

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/presenter.ts [hunk 1]
```diff
@@ -29,6 +33,17 @@ export const postPresenterPage = async (req: Request, res: Response, next: NextF
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
 
     const session = req.session as Session;
+
+    if (isRemoveJourney(req)) {
+      const appData: ApplicationData = getApplicationData(session);
+      if (!appData[Transactionkey]) {
+        const transactionID = await postTransaction(req, session);
+        appData[Transactionkey] = transactionID;
+        appData[IsRemoveKey] = true;
```
Comment (mwestacott) [inline src/utils/presenter.ts:42]: We've recently talked about moving this setting of the 'is_remove' field earlier in the journey but leaving it here for now.

Comment (markpit) [reply src/utils/presenter.ts:42]: looking at update.filing.controller, it looks like we are missing a call to 
      setExtraData(req.session, appData);
to save the modified appData (with transaction ids) back to the session.

Comment (dlloyd1-ch) [reply src/utils/presenter.ts:42]: Will unit tests be added before this move out of draft?

Comment (mwestacott) [reply src/utils/presenter.ts:42]: Yep, that's my plan.

I'll add in that call to `setExtraData()` too.

Comment (mwestacott) [reply src/utils/presenter.ts:42]: Unit tests and that missing call now added.

**Review by dlloyd1-ch — APPROVED** [review]
Looks good to me.

File: test/controllers/update/overseas.entity.presenter.controller.spec.ts [hunk 2]
```diff
@@ -184,5 +198,24 @@ describe("OVERSEAS ENTITY PRESENTER controller", () => {
...
+        `${OVERSEAS_ENTITY_PRESENTER_URL}${JOURNEY_REMOVE_QUERY_PARAM}`).send(PRESENTER_OBJECT_MOCK);
+
+      expect(resp.status).toEqual(302);
+
+      expect(mockData[Transactionkey]).toEqual(TRANSACTION_ID);
+      expect(mockData[OverseasEntityKey]).toEqual(OVERSEAS_ENTITY_ID);
+      expect(mockData[IsRemoveKey]).toEqual(true);
+      expect(mockTransactionService).toHaveBeenCalledTimes(1);
+      expect(mockCreateOverseasEntity).toHaveBeenCalledTimes(1);
+      expect(mockSaveAndContinue).toHaveBeenCalledTimes(1);
```
Comment (markpit) [inline test/controllers/update/overseas.entity.presenter.controller.spec.ts:216]: i would add an expect to check that mockSetExtraData is called, as that is what updates the appData  in the session. You might have to declare the mockSetExtraData as I don't think it is used currently.

Comment (mwestacott) [reply test/controllers/update/overseas.entity.presenter.controller.spec.ts:216]: 👍 will do.

Comment (mwestacott) [reply test/controllers/update/overseas.entity.presenter.controller.spec.ts:217]: New assertion added.

**Review by markpit — APPROVED** [review]

**Pull request merged**